### ISSUE TITLE
[prototype] ANet 02 — reproduce the ActivityNet baseline

### ISSUE THREAD

**Issue by ulassbin** [ulassbin/CORAL#86]
---
type: prototype
blocked_by: [01]
---
# Reproduce the ActivityNet baseline

**FRONTIER — takeable now.** Ticket 01 resolved 2026-08-13.

⚠️ **Governed by `docs/ANET_PROTOCOL_v1.md`**, frozen AND reviewed (Ulaş's ruling is §14 **A1**).
Requirements this ticket did not originally carry, added by 01 and A1:

1. **Run TWO baselines, never conflated** (§5): a **port validation** on all 4,819 training videos,
   evaluated on val against CoLA's ≈26 — this is the single pre-authorised *calibration* read of
   §3 — and the **anchor** on the 4,019 videos left after the dev-800 holdout
   (`docs/anet_dev_split_v1.json`), evaluated on dev. Without both, a low number is ambiguous
   between "our adapter is broken" and "the holdout cost us".
2. **This ticket produces four numbers the protocol left deliberately unmeasured**, all landing as
   amendment **A2** before ticket 03 runs: the **discordance s** and the **paired design effect**
   (§9's MDE table), the **practical effect δ** — which A1 defines as
   **0.05 × (1.0 − anchor mean per-instance best-IoU)**, not the 0.02 v1 proposed — and the
   **operating dev-split size**. The reference proposal budget in dets/video lands as **A3** (§11).

3. **Choose the dev-split size from the nested candidates** (A1). `docs/anet_dev_split_v1.json`
   carries **400 ⊂ 600 ⊂ 800**, so one baseline run prices all three. The rule, pre-declared and
   not revisable afterwards: take the **smallest** k whose 1.5×MDE on ALL is at or below the
   measured δ, provided the holdout cost — the gap between the two baselines above — is within its
   own noise; if none qualifies, keep 800 and record the shortfall.
4. **Budget: 1 port-validation run + 5 anchor runs** (seeds 1111–5555). **No sweeps** — every
   constant declared before the first run. soft-NMS σ and all THUMOS-tuned constants are *not*
   inherited; declare σ before running and report it.

## Question
Stand up a CoLA-family generator on ANet-1.2 and reproduce a credible baseline against the published
reference (**CoLA ANet-1.2 ≈ 26 avg mAP**, tIoU 0.5:0.05:0.95).

## Notes for whoever picks this up
- **Everything needed is on disk; zero downloads** — `ActivityNet1.2-I3D-JOINTFeatures.npy` (3.4 GiB),
  `ActivityNet1.2-Annotations/`, `gt.json`, `anet_create_json.py` under
  `~/datasets/action_recognition_datasets/activitynet_1.2/`. **`np.load` needs `encoding='latin1'`.**
- **A dataset adapter must be WRITTEN, not configured.** `core/config.py` and `core/dataset.py`
  hard-code THUMOS paths and `NUM_CLASSES=20`; the W-TALC `.npy`-blob layout differs from our
  per-video layout; eval must switch to tIoU 0.5:0.05:0.95. Stock CoLA released no ANet code to crib.
  The #51 audit estimated 1–2 days for this adapter alone.
- **Expect the validated THUMOS constants not to hold.** `lr`, `batch_size`, `contrast_rampup_steps`,
  adaptive stopping and soft-NMS σ=0.10 were all tuned on 200 videos; ANet has ~4.8k. The *principles*
  port, the constants do not.
- A reproduction that lands materially below ≈26 is a finding about the port, not about ActivityNet —
  say which before moving on.

## Done when
A baseline runs end to end under the frozen protocol, with its number reported beside the published
reference and any deviation explained.

**Comment by ulassbin** [comment]
**RESOLVED 2026-08-17.** A CoLA-family generator runs end to end on ActivityNet-1.2 and reproduces
the published reference. Full ledger: `docs/ANET_official_anet12_audit.md`.

## The number, beside the published reference

CoLA's official `anet12` code, its official 4,819/2,383 split, its recipe verbatim (batch 256,
`NUM_SEGMENTS=50`, lr 1e-4 for 100k, `R_EASY`/`R_HARD` 10/8, `CLASS_THRESH` 0.1, `NMS_THRESH` 0.6,
SniCo 0.005, best-checkpoint on validation — the same selection rule CoLA itself uses, so the
comparison is like-for-like on that axis), changing **exactly one thing**:

| run | @0.5 | @0.75 | @0.95 | **AVG** | vs CoLA ≈26 |
|---|---:|---:|---:|---:|---:|
| official recipe, unmodified, full 100k steps | 0.0143 | — | 0.0027 | **1.04** | −25 |
| official recipe **+ `ACTION_SCALE=10`** | 39.09 | 25.48 | 2.88 | **24.72** | −1.3 |
| best config: **+ `NUM_SEGMENTS=200`** | 44.46 | 27.34 | 4.05 | **27.25** | +1.25 |

## The one deviation, explained

`core/loss.py:ActionLoss` is `nn.BCELoss()` with mean reduction, so it divides by the class count.
CoLA balanced every other loss weight against that divisor at C=20; at C=100 the classification
gradient is ~11× weaker (measured f_cls grad L2 **2.30e-2** THUMOS vs **2.10e-3** ActivityNet) while
the SniCo branch is untouched. Left alone, `Loss/Action` never leaves the uniform 100-way softmax
value (0.05598 → 0.05499 across the **full 100,000 steps**) and avg mAP is 1.04. **"It just needed
more steps" is excluded by that run.** `ACTION_SCALE` rescales the action term only, at CoLA's own
lr — chosen over raising lr, which would rescale the contrastive branch too. Declared as a constant
in **A4**, with the 5/10/20 comparison and its out-of-authorisation status disclosed there.

## How it was diagnosed (what was ruled out, so it is not re-run)

Falsified: channel order (ANet `[:1024]`=rgb, `[1024:]`=flow, signatures match our THUMOS features)
· feature provenance · fps / `t_factor` (confirmed three ways; T/duration 1.5565 vs 25/16=1.5625)
· the decoder · linear LR scaling (diverged) · batch size alone (0.0331) · dropout and all 27 config
constants · "train longer". Positive controls: a 100-video train==test probe reached accuracy 1.000
/ mAP 0.3977, and a 1,000-video probe covering all 100 classes fit with **no intervention** at mAP
0.2545 — so the machinery is sound at 1,000 videos and fails only at 4,819. That localised the
fault to gradient balance rather than plumbing.

## Two retractions this ticket owns

- **A3** — the premise "stock CoLA released no ActivityNet code" was **false**; the `anet12` branch
  exists. Ticket 02 propagated it unchecked. The earlier **13.27 / 17.27** are reclassified as
  *inferred-port* results and may not be tabled against ≈26.
- **The `NUM_SEGMENTS` conclusion** — "more segments is a straight loss" was drawn from a run pinned
  at initialisation, which measured the training failure, not resolution. Retracted; once training
  works, resolution is monotonically helpful to at least 200 (**A6**).

## Residual caveats, carried forward

- **Features are not CoLA's own release.** Ours are the W-TALC/CO2-Net JOINT blob; CoLA's ANet-1.2
  features are a permission-walled Drive download. Same 25 fps / stride-16 convention (measured),
  but a different extraction. A3's standing instruction — no anchor runs until the official recipe
  runs on the official features — is **discharged by A5 retiring the anchor runs**, not by the
  condition being met. Every ≈26 comparison carries this caveat.
- **All three numbers above are best-checkpoint-selected** (~40 validation evaluations per run, max
  kept). Fine for a reproduction against CoLA, which selects the same way; **not** fine for an
  ablation ladder, which is why **A5** changes the regime to fixed-step for everything downstream.
- Single seed (0). Within-run oscillation **±0.5 avg mAP** — larger than the 100↔200 gap.

## Not done, and deliberately so

The five anchor runs and the trainA-4019 → dev-800 design were never spent: **A5** supersedes that
structure for the ablation programme. Consequently the four deferred measurements A1 assigned to
this ticket — discordance **s**, the paired design effect, **δ = 0.05 × (1 − anchor mean best-IoU)**
and the operating split size **k** — and §11's reference proposal budget in dets/video are **not
produced here**. They belong to a dev-side gating design that this programme no longer runs; if a
gated dev-side rung is ever revived, they must be measured then. Recorded so the omission is visible
rather than silent.

**Amendments landed:** A4 (`ACTION_SCALE` declared) · A5 (fixed-step ablation regime, and its
disclosure obligation) · A6 (A2's screen superseded, `NUM_SEGMENTS=200` frozen).

**Unblocks:** ticket 03. The diagnosis runs read-only on this baseline's pool — no new training.

**Comment by ulassbin** [comment]
Resolved — see the Answer above. Unblocks #87, which is now the frontier.
